src/librustc_typeck/coherence/mod.rs

```diff
@@ -250,29 +250,44 @@ impl<'a, 'gcx, 'tcx> CoherenceChecker<'a, 'gcx, 'tcx> {
 
             match param_env.can_type_implement_copy(tcx, self_type, span) {
                 Ok(()) => {}
-                Err(CopyImplementationError::InfrigingField(name)) => {
-                    struct_span_err!(tcx.sess,
-                                     span,
-                                     E0204,
-                                     "the trait `Copy` may not be implemented for this type")
-                        .span_label(span, &format!("field `{}` does not implement `Copy`", name))
-                        .emit()
+                Err(CopyImplementationError::InfrigingField(did, name)) => {
+                    let mut err = struct_span_err!(tcx.sess,
+                                                   span,
+                                                   E0204,
+                                                   "the trait `Copy` may not be implemented for \
+                                                   type `{}`",
+                                                   self_type);
+                    err.span_label(span,
+                                   &format!("field `{}` in struct `{}` doesn't implement `Copy`",
```

Comment:
Even when the span for the field definition is not available, this `span_label` will [point at the `impl Copy for Struct`](https://github.com/rust-lang/rust/pull/37493/files#diff-fbd95c2e54e6b95eb365ba6cdde87679). Because of this, I changed it to be `field 'item' doesn't implement 'Copy'` always.